### PR TITLE
Mes 6456 loading spinner error handling

### PR DESCRIPTION
**Pull request by tomcrawleyy**
## Description
Introduces logic for the loading spinner and error handling
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [x] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image

### CODE REVIEW

File: src/app/pages/journal/journal.page.ts [hunk 1]
```diff
@@ -204,36 +210,33 @@ export class JournalPage extends BasePageComponent implements OnInit {
...
+  };
 
   showError = (error: MesError): void => {
     if (error === undefined || error.message === '') return;
-
     // Modals are at the same level as the ion-nav so are not getting the zoom level class,
     // this needs to be passed in the create options.
 
-    // TODO figure out what to do about zoom methods not existing in appComponent
-    // const zoomClass = `modal-fullscreen ${this.app.getTextZoomClass()}`;
```
Comment (matthew2564) [inline src/app/pages/journal/journal.page.ts:230]: `this.app.getTextZoomClass()` should now have been implemented so we should be able to use that here

**Comment by matthew2564** [comment]
On the Screenshot looks to need a space added between `journal` & `and`

**Comment by andrewsetterfield** [comment]
> On the Screenshot looks to need a space added between `journal` & `and`

added a space

File: src/components/common/error-message/error-message.module.ts [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
+//   ],
+//   imports: [
+//     IonicModule,
+//   ],
+//   providers: [],
+//   exports: [
+//     ErrorMessageComponent,
+//   ],
+// })
+// export class ErrorMessageModule { }
```
Comment (tomcrawleyy) [inline src/components/common/error-message/error-message.module.ts]: this can be deleted now

Comment (andrewsetterfield) [reply src/components/common/error-message/error-message.module.ts]: i commented it out to confirm it wasnt needed. good spot, removed now.

File: mock/ionic-mocks/modal-controller.mock.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export class ModalControllerMock {
+  dismiss = () => {}
```
Comment (matthew2564) [inline mock/ionic-mocks/modal-controller.mock.ts]: dismiss returns a Promise boolean

File: src/app/pages/journal/journal.page.ts [hunk 4]
```diff
@@ -72,7 +74,7 @@ export class JournalPage extends BasePageComponent implements OnInit {
 
   pageState: JournalPageState;
   selectedDate: string;
-  // loadingSpinner: Loading;
+  loadingSpinner$: Observable<HTMLIonLoadingElement>;
```
Comment (matthew2564) [inline src/app/pages/journal/journal.page.ts:77]: is this used?

**Review by matthew2564 — APPROVED** [review]

**Review by Cullima — APPROVED** [review]

**Pull request merged**